sdk/nodejs/kinesis/kinesisMixins.ts

```diff
@@ -26,6 +27,38 @@ export interface StreamEventSubscriptionArgs {
      */
     readonly batchSize?: number;
 
+    /**
+     * If the function returns an error, split the batch in two and retry. Defaults to `false`.
+     */
+    readonly bisectBatchOnFunctionError?: boolean;
+
+    /**
+     * An Amazon SQS queue or Amazon SNS topic destination for failed records.
+     */
+    readonly destinationConfig?: pulumi.Input<types.input.lambda.EventSourceMappingDestinationConfig>;
+
+    /**
+     * The maximum amount of time to gather records before invoking the function, in seconds. Records will continue to buffer
+     * until either maximum_batching_window_in_seconds expires or batch_size has been met. Defaults to as soon as records
+     * are available in the stream. If the batch it reads from the stream only has one record in it, Lambda only sends one record to the function.
+     */
+    readonly maximumBatchingWindowInSeconds?: number;
+
+    /**
+     * The maximum age of a record that Lambda sends to a function for processing. Minimum of `60`, maximum and default of `604800`
+     */
+    readonly maximumRecordAgeInSeconds?: number;
+
+    /**
+     * The maximum number of times to retry when the function returns an error. Minimum of `0`, maximum and default of `10000`.
+     */
+    readonly maximumRetryAttempts?: number;
+
+    /**
+     * The number of batches to process from each shard concurrently. Minimum and default of `1`, maximum of `10`
+     */
+    readonly parallelizationFactor?: number;
+
```

Comment:
So do we want `startingPositionTimestamp` in Kinesis?

Reply:
It's already in there ;)

Reply:
Oh, f*ck, sorry